pkg/store/proxy.go

```diff
@@ -242,13 +239,13 @@ func (s *ProxyStore) LabelValues(ctx context.Context, r *storepb.LabelValuesRequ
 	)
 	stores, err := s.stores(ctx)
 	if err != nil {
-		return nil, grpc.Errorf(codes.Unknown, err.Error())
+		return nil, status.Errorf(codes.Unknown, err.Error())
 	}
-	for _, s := range stores {
+	for _, st := range stores {
 		wg.Add(1)
-		go func(s *Info) {
+		go func(s Client) {
 			defer wg.Done()
-			resp, err := s.Client.LabelValues(ctx, &storepb.LabelValuesRequest{
+			resp, err := st.LabelValues(ctx, &storepb.LabelValuesRequest{
```

Comment:
I think this should be `s.LabelValues` since that's what's passed as an argument to this function (otherwise, `s` would be unused and only the first store in `stores` will be contacted).

Reply:
yup! That would be a major bug. Thanks!